### PR TITLE
Introducing philips hue colors switching using commands api

### PR DESCRIPTION
**Pull request by andrenatal**
@tamarahills @hobinjk could you guys please review this?

### CODE REVIEW

File: src/controllers/commands_controller.js [hunk 1]
```diff
@@ -114,28 +114,48 @@ CommandsController.post('/', function (request, response) {
...
               let name = obj.name.toUpperCase();
               if (name == match) {
-                payload.href = obj.properties.on.href;
+                thingfound = true;
                 break;
               }
             }
-            if (payload.href != '') {
+            if (thingfound) {
               // Returning 201 to signify that the command was mapped to an
```
Comment (hobinjk) [inline src/controllers/commands_controller.js]: nit: can you move this comment to where it returns 201?

File: src/controllers/commands_controller.js [hunk 2]
```diff
@@ -114,28 +114,48 @@ CommandsController.post('/', function (request, response) {
...
-              iotOptions.uri = CommandsController.gatewayHref + payload.href;
+              if ((payload.param3 != null) && 
+                (payload.param3 != ''  && obj.properties.hue != '')) {
+                var colorname_to_hue = {red:0, orange:8500, yellow:17000, 
+                  green:25500, white:34000, blue:46920, purple:48000, 
+                  magenta:54000, pink:60000};
+                if (!colorname_to_hue[payload.param3]) {
+                  response.status(404).send(JSON.stringify(
+                    {'message': 'Hue color not found'}));
+                    return;
```
Comment (hobinjk) [inline src/controllers/commands_controller.js]: nit: this should be indented one less level

File: src/controllers/commands_controller.js [hunk 3]
```diff
@@ -114,28 +114,48 @@ CommandsController.post('/', function (request, response) {
...
-              response.status(201).send(JSON.stringify(
-                {'message': 'Command Created'}));
-              iotOptions.uri = CommandsController.gatewayHref + payload.href;
+              if ((payload.param3 != null) && 
+                (payload.param3 != ''  && obj.properties.hue != '')) {
+                var colorname_to_hue = {red:0, orange:8500, yellow:17000, 
+                  green:25500, white:34000, blue:46920, purple:48000, 
+                  magenta:54000, pink:60000};
+                if (!colorname_to_hue[payload.param3]) {
+                  response.status(404).send(JSON.stringify(
```
Comment (hobinjk) [inline src/controllers/commands_controller.js]: nit: `response.status(404).json({that object})` would be simpler than calling JSON.stringify

File: src/controllers/commands_controller.js [hunk 4]
```diff
@@ -114,28 +114,48 @@ CommandsController.post('/', function (request, response) {
...
+                  return;
+              }
+              iotOptions.uri = CommandsController.gatewayHref + payload.href;  
               iotOptions.headers.Authorization = 'Bearer ' +
-                CommandsController.jwt;
-
-              (payload.param2 == 'on')?
-                iotOptions.body = JSON.stringify({'on': true}):
-                iotOptions.body = JSON.stringify({'on': false});
+              CommandsController.jwt;
```
Comment (hobinjk) [inline src/controllers/commands_controller.js]: nit: this should be indented

File: src/controllers/commands_controller.js (hunk 3 above)
Comment (andrenatal) [reply src/controllers/commands_controller.js]: fixed

File: src/controllers/commands_controller.js (hunk 1 above)
Comment (andrenatal) [reply src/controllers/commands_controller.js]: fixed

File: src/controllers/commands_controller.js (hunk 2 above)
Comment (andrenatal) [reply src/controllers/commands_controller.js]: fixed

File: src/controllers/commands_controller.js (hunk 4 above)
Comment (andrenatal) [reply src/controllers/commands_controller.js]: fixed

**Comment by andrenatal** [comment]
Thank you guys. I'll file an issue ok @hobinjk ?

**Comment by hobinjk** [comment]
@andrenatal Sounds good to me

**Pull request merged**